weave/legacy/weave/ref_util.py

```diff
@@ -0,0 +1,106 @@
+import dataclasses
```

Comment:
not legacy

Reply:
I think there is some dependency on the refs code in query service. There is a duplicate file at `trace/ref_util.py` for the trace code.